### PR TITLE
Restore the family value in os-release

### PR DESCRIPTION
**Pull request by Itxaka**
Probably fixes https://github.com/kairos-io/kairos/issues/2387

### CODE REVIEW

**Comment by Itxaka** [comment]
Confirmed that it fixes https://github.com/kairos-io/kairos/issues/2387

File: versioneer/cli.go [hunk 1]
```diff
@@ -97,6 +97,13 @@ var (
 		Usage:   "the url where more information about the project can be found",
 		EnvVars: []string{EnvVarHomeURL},
 	}
+
+	familyFlag *cli.StringFlag = &cli.StringFlag{
+		Name:    "family",
+		Value:   "",
+		Usage:   "the url where more information about the project can be found",
```
Comment (jimmykarily) [inline versioneer/cli.go:104]: this is not correct (copy paste error?)

**Comment by jimmykarily** [comment]
The "family" has an 1-1 relation with the Dockerfile being used to build the image. Restructuring the dockerfiles shouldn't have any effect on anything.

I believe that whatever code is relying on "family" to do things, should rather rely on another field (flavor?).

In other words, instead of adding family back, we should change the code that relies on this, to not rely.

File: versioneer/cli.go (hunk 1 above)
Comment (Itxaka) [reply versioneer/cli.go:104]: Indeed

**Comment by Itxaka** [comment]
> The "family" has an 1-1 relation with the Dockerfile being used to build the image. Restructuring the dockerfiles shouldn't have any effect on anything.
> 
> I believe that whatever code is relying on "family" to do things, should rather rely on another field (flavor?).
> 
> In other words, instead of adding family back, we should change the code that relies on this, to not rely.

I don't understand the restructuring the docker files means in this context.

We were relaying in those values to identify the family of the booted system, in this case for selinux, but could be anything else in order to deal with quirks. 

When moving to this, it stopped working. We didn't noticed because it just made selinux disabled everywhere lol

And yes, we can move to a different test in order to fix this but this is still a regression IMO. Changing the test in the bootargs is much more complex than this as it requires inserting modules like regex, which already gave us issues as they were not signed for secureboot and such.


IMO fixing the regression first and then moving to improve the actual bootargs if we want to drop the family is the best and fastest approach on this.

**Comment by jimmykarily** [comment]
> > The "family" has an 1-1 relation with the Dockerfile being used to build the image. Restructuring the dockerfiles shouldn't have any effect on anything.
> > I believe that whatever code is relying on "family" to do things, should rather rely on another field (flavor?).
> > In other words, instead of adding family back, we should change the code that relies on this, to not rely.
> 
> I don't understand the restructuring the docker files means in this context.
> 
> We were relaying in those values to identify the family of the booted system, in this case for selinux, but could be anything else in order to deal with quirks.
> 
> When moving to this, it stopped working. We didn't noticed because it just made selinux disabled everywhere lol
> 
> And yes, we can move to a different test in order to fix this but this is still a regression IMO. Changing the test in the bootargs is much more complex than this as it requires inserting modules like regex, which already gave us issues as they were not signed for secureboot and such.
> 
> IMO fixing the regression first and then moving to improve the actual bootargs if we want to drop the family is the best and fastest approach on this.

Agreed. I'm just trying to say that "family" was just another way to say "which dockerfile is being used at build time" and we shouldn't add more meaning to it. It shouldn't be used after building is done. Mainly because it doesn't add any new information on top of flavor/flavor-release.

Let's fix it like this for now since it's a regression but if you agree, we can try to change the dependent code to not rely on this anymore.

**Comment by jimmykarily** [comment]
> > > > The "family" has an 1-1 relation with the Dockerfile being used to build the image. Restructuring the dockerfiles shouldn't have any effect on anything.
> > > > I believe that whatever code is relying on "family" to do things, should rather rely on another field (flavor?).
> > > > In other words, instead of adding family back, we should change the code that relies on this, to not rely.
> > > 
> > > 
> > > I don't understand the restructuring the docker files means in this context.
> > > We were relaying in those values to identify the family of the booted system, in this case for selinux, but could be anything else in order to deal with quirks.
> > > When moving to this, it stopped working. We didn't noticed because it just made selinux disabled everywhere lol
> > > And yes, we can move to a different test in order to fix this but this is still a regression IMO. Changing the test in the bootargs is much more complex than this as it requires inserting modules like regex, which already gave us issues as they were not signed for secureboot and such.
> > > IMO fixing the regression first and then moving to improve the actual bootargs if we want to drop the family is the best and fastest approach on this.
> > 
> > 
> > Agreed. I'm just trying to say that "family" was just another way to say "which dockerfile is being used at build time" and we shouldn't add more meaning to it. It shouldn't be used after building is done. Mainly because it doesn't add any new information on top of flavor/flavor-release.
> > Let's fix it like this for now since it's a regression but if you agree, we can try to change the dependent code to not rely on this anymore.
> 
> Ahh gotcha. Indeed, I still think it's useful as it covers a more generic way for us to identify and apply quirks if needed, i.e. rhel family allows us to cover all rhel, rocky Linux, alma Linux, etc.. which to do with existing keys it's not that nice.
> 
> Still yeah, it should be dropped if the meaning behind it was just to know the docker files to be used on building, I did not know that I thought we were just grouping them in a nicer global way!

@mauromorales might argue that family has a deeper meaning, I'm not 100% sure. Maybe the family has grown to something more important nowadays :D.

**Comment by mauromorales** [comment]
IMO the family is not related to the dockerfile per se, but to how we can use the same approach for a group of distributions. Maybe at some point we can merge debian and ubuntu into one, or maybe it makes more sense to split rhel into fedora and rhel "replicas" (for lack of a better word). At the end of the day is a way to try to DRY as much as possible logic in the dockerfiles because whenever we were doing updates before, we constantly forgot to do the updates for other distributions/releases.

**Review by mauromorales — APPROVED** [review]

**Pull request merged**